File: 051/9.py

class BookNotAvailableException(Exception):
    def __init__(self, message):
        super().__init__(message)


class Book:
    def __init__(self, title, author, isbn):
        self.title = title
        self.author = author
        self.isbn = isbn
        self.isAvailable = True  

 
    def borrowBook(self):
        if self.isAvailable:
            self.isAvailable = False
            print(f"คุณยืมหนังสือ: {self.title} ของ {self.author} (ISBN: {self.isbn})")
        else:
            raise BookNotAvailableException(f"หนังสือ '{self.title}' ถูกยืมอยู่แล้ว!")

 
    def returnBook(self):
        if not self.isAvailable:
            self.isAvailable = True
            print(f"คุณคืนหนังสือ: {self.title} ของ {self.author} (ISBN: {self.isbn})")
        else:
            print(f"หนังสือ '{self.title}' ยังไม่ได้ถูกยืม")


    def displayDetails(self):
        status = "พร้อมใช้งาน" if self.isAvailable else "ถูกยืมอยู่"
        print(f"ชื่อหนังสือ: {self.title}")
        print(f"ผู้แต่ง: {self.author}")
        print(f"ISBN: {self.isbn}")
        print(f"สถานะ: {status}")


try:
    book1 = Book("Python for Beginners", "John Doe", "978-3-16-148410-0")
    
    book1.displayDetails()  
    book1.borrowBook()      
    book1.displayDetails()  
    book1.borrowBook()
except BookNotAvailableException as e:
    print(f"ข้อผิดพลาด: {e}")

print("\n")


book1.returnBook()
book1.displayDetails()  

File: 051/9.js

class BookNotAvailableException extends Error {
    constructor(message) {
        super(message);
        this.name = "BookNotAvailableException";
    }
}


class Book {
    constructor(title, author, isbn) {
        this.title = title;
        this.author = author;
        this.isbn = isbn;
        this.isAvailable = true;  
    }

    borrowBook() {
        if (this.isAvailable) {
            this.isAvailable = false;
            console.log(`คุณยืมหนังสือ: ${this.title} ของ ${this.author} (ISBN: ${this.isbn})`);
        } else {
            throw new BookNotAvailableException(`หนังสือ '${this.title}' ถูกยืมอยู่แล้ว!`);
        }
    }


    returnBook() {
        if (!this.isAvailable) {
            this.isAvailable = true;
            console.log(`คุณคืนหนังสือ: ${this.title} ของ ${this.author} (ISBN: ${this.isbn})`);
        } else {
            console.log(`หนังสือ '${this.title}' ยังไม่ได้ถูกยืม`);
        }
    }


    displayDetails() {
        const status = this.isAvailable ? "พร้อมใช้งาน" : "ถูกยืมอยู่";
        console.log(`ชื่อหนังสือ: ${this.title}`);
        console.log(`ผู้แต่ง: ${this.author}`);
        console.log(`ISBN: ${this.isbn}`);
        console.log(`สถานะ: ${status}`);
    }
}


try {
    const book1 = new Book("Python for Beginners", "John Doe", "978-3-16-148410-0");
    
    book1.displayDetails();  
    book1.borrowBook();      
    book1.displayDetails();  
    
    
    book1.borrowBook();
} catch (error) {
    if (error instanceof BookNotAvailableException) {
        console.log(`ข้อผิดพลาด: ${error.message}`);
    } else {
        console.log("ข้อผิดพลาดที่ไม่คาดคิด:", error);
    }
}

console.log("\n");


const book2 = new Book("JavaScript for Beginners", "Jane Doe", "978-0-12-345678-9");
book2.borrowBook();  
book2.returnBook();  
book2.displayDetails();  
